### DISCUSSION TITLE
Statamic "blade directives" addon issue / better implementation?

### DISCUSSION THREAD

**Discussion by torian257x** [statamic/cms#4487] [Q&A]
I encountered this issue here:
https://github.com/edalzell/statamic-blade/issues/68

now edalzell does not seem very active (closed the issue I referenced after months? without a comment)

in short, his addon `json_serialize`s everything that can be json serialized in the view if you use his ViewServiceProvider. Including eloquent models, Enums and I am sure, a lot more.

that is a bit unexpected, I expected his addon to make it easier to use statamic values, not that he messes with data that isn't statamic. 

Is there another "check" that one could use instead of `$data instanceof JsonSerializable` that would lead to a more expected behaviour and work any way?

like, before I start coming up with my own checks, I thought I'd rather ask the experts for some advice.

**Comment by jasonvarga** [comment]
This isn't really the place to be asking about third party addons. It looks like he fixed the issue you referenced. Did you update?

**Reply by torian257x** [reply]
ahh now I see, didn't notice his commit, thanks a lot for the pointer. 

Seems like he hasn't released a new version....

But I don't seem to be able to see why this would solve the problem. It only restricts the problem now to statamic routes.
So slightly better (?), but feels a bit more like a "patchwork" fix than a proper fix. Just imagine, you use a template once via statamic, once via normal route. Gives you completely different results now. Is that better? probably actually worse.


3rd party addons>
[I read](https://statamic.com/forum/5618-why-antlers#reply-17922) that you are

> going to elevate Blade with some love and attention

if it helps I can reformulate the question so you cannot argue 3rd party addon 😇 

but it's a bit childish, no? "I have data, please gimme easy to use data with values"
